### PR TITLE
Update of CES mark-up cost implementation: industry mark-up cost budget-neutral as tax, adapt some default cost, refactoring

### PR DESCRIPTION
**Pull request by fschreyer**
# Purpose of this PR

This provides an update to the CES mark-up cost formulation in buildings and industry as discussed [in this issue](https://github.com/remindmodel/development_issues/issues/119). The table in the issue provides the details on which cost have been changed and why. 

Main changes are:

* CES mark-up cost in industry now implemented as tax and not as cost to the energy system which have an impact on GDP. This was done as these cost do not correspond to real-world investment cost. 
* CES mark-up cost in industry on H2 and electrification nodes were altered to improve substitution behavior in high Elec/H2 runs 

This change requires recalibration of the CES parameters. I could provide such a set of parameters. As we intend to merge some other changes to the CES function, though, it might be worth to merge them all at once and recalibrate only afterwards. 

## Type of change

(Make sure to delete from the Type-of-change list the items not relevant to your PR)

- [x] Refactoring
- [x] Minor change (default scenarios show only small differences)

## Checklist:

- [x] My code follows the coding etiquette
- [x] I have performed a self-review of my own code
- [x] Changes are commented, particularly in hard-to-understand areas
- [x] I have updated the in-code documentation
- [x] The model compiles and runs successfully (`Rscript start.R -q`)
- [ ] wait for other PRs that change CES function -> common calibration

## Further information (optional):

* Test runs are here: 
* Comparison of results (what changes by this PR?): 

REMIND 12 regions runs with above cost update and Elec or H2 focus:

https://cloud.pik-potsdam.de/index.php/apps/files/?dir=/Shared/REMIND-EU_calibration/Results/ElecH2_2022_08_29&openfile=14279718

### CODE REVIEW

File: modules/36_buildings/simple/datainput.gms [hunk 1]
```diff
@@ -92,19 +92,40 @@ if ((cm_ElLim_b lt 1),
 );
 
 
-*** FS: CES markup cost buildings
+*** CES mark-up cost buildings
```
Comment (LaviniaBaumstark) [inline modules/36_buildings/simple/datainput.gms:95]: parts of this documentation should also show up in the automated documentation. Please, use "*`" for those parts

File: modules/37_industry/subsectors/datainput.gms [hunk 2]
```diff
@@ -317,32 +317,47 @@ pm_ue_eff_target("ue_otherInd")         = 0.008;
 
 
 
-*** FS: CES markup cost industry
-*** default values of CES markup
-p37_CESMkup(t,regi,in) = 0;
+*** CES mark-up cost industry
```
Comment (LaviniaBaumstark) [inline modules/37_industry/subsectors/datainput.gms:320]: also this comments might go into the automated documentation. If they are the same for all realizations they should go to the module description

**Review by 0UmfHxcvx5J7JoaOhFSs5mncnisTJJ6q — APPROVED** [review]

**Comment by fschreyer** [comment]
This is ready to merge but we will wait for some other PRs that change the elasticity of the CES function to calibrate all in one.

File: modules/36_buildings/simple/datainput.gms (hunk 1 above)
Comment (fschreyer) [reply modules/36_buildings/simple/datainput.gms]: done

File: modules/37_industry/subsectors/datainput.gms (hunk 2 above)
Comment (fschreyer) [reply modules/37_industry/subsectors/datainput.gms]: I agree. The module and realization descriptions are quite short and would need some extension anyways. Let's tackle this in a future cleaning session. I created this [issue](https://github.com/remindmodel/development_issues/issues/157).

**Comment by fschreyer** [comment]
New CES parameters for SSP2 and SSP1 uploaded together with #994. Ariadne scenario config updated for new CES mark-up cost values used for electrification, hydrogen and synfuel scenarios.

**Pull request merged**